### ISSUE TITLE
Extremely slow on some websites

### ISSUE THREAD

**Issue by logan2k** [mullvad/mullvadvpn-app#3773]
## Issue report

Operating system: Debian 11

App version: 2022.2

### Issue description

I found some website will be really slow when using mullvad VPN.

Normal speed (good websites list below, or using other VPNs on bad websites) is 200KB/s - 2MB/s, but using Mullvad on bad websites it's 3-30KB/s, most time they will never finish loading.

It don't become better with different server/region, neither with different browsers.

Known "bad" websites:
https://www.speedtest.net/
https://www.reddit.com/
https://www.ebay.com/ (only sometimes)
https://duckduckgo.com/ (only sometimes)

Known "good" websites:
https://www.google.com/
https://www.youtube.com/
https://www.wikipedia.org/
https://www.debian.org/
https://www.facebook.com/
https://www.microsoft.com/

How can I solve this?

**Comment by vvaadim** [comment]
try playing with your wireguard adapter MTU value, or try using OpenVPN protocol with stock mullvad settings.

**Comment by logan2k** [comment]
> try playing with your wireguard adapter MTU value, or try using OpenVPN protocol with stock mullvad settings.

Thanks.
Changing MTU don't help the issue.
OpenVPN barely connect because of the sensorship in the country I current at.

**Comment by vvaadim** [comment]
> > try playing with your wireguard adapter MTU value, or try using OpenVPN protocol with stock mullvad settings.
> 
> Thanks. Changing MTU don't help the issue. OpenVPN barely connect because of the sensorship in the country I current at.

So it might be related to their DPI detection of packets, try using wireguard udp2tcp via Mullvad cli. Also you can also try selecting OpenVPN TCP in settings and selecting a "normal" port like 443.

**Comment by githubdotcomaccount** [comment]
I'm seeing this same issue on Android (GrapheneOS) as well as my desktop (Fedora 36). I'm on the newest version of the apps for each. Not sure if I can implement the suggestion above on Android at all or how to do so on Fedora. The server I'm on for my desktop is a WireGuard server (tried a few), as is my phone. A couple of users on Reddit also mentioned this (https://libredd.it/r/mullvadvpn/comments/wadsl9/mullvad_is_constantly_failing_since_yesterday/) though using WireGuard server seems to have resolved it for them, strangely.

**Comment by logan2k** [comment]
> > > try playing with your wireguard adapter MTU value, or try using OpenVPN protocol with stock mullvad settings.
> > 
> > 
> > Thanks. Changing MTU don't help the issue. OpenVPN barely connect because of the sensorship in the country I current at.
> 
> So it might be related to their DPI detection of packets, try using wireguard udp2tcp via Mullvad cli. Also you can also try selecting OpenVPN TCP in settings and selecting a "normal" port like 443.

I tried both your suggestions, sadly none works.

**Comment by logan2k** [comment]
I just waited long enough so speedtest.net would finish loading, the results is in the picture.
Please note the slow download speed shown, the real world speed is even much slower.
![Uploading slow.png…]()

**Comment by logan2k** [comment]
Today I just found server "ca28-wireguard" don't have this problem.
There must be some config difference on it.

**Comment by faern** [comment]
>  OpenVPN barely connect because of the sensorship in the country I current at.

If you live somewhere with known censorship I think that's the issue. All our servers have almost identical configuration. So I don't think it's the configuration. Something else that will impact your bandwidth and latency to a given server is the peering. In other words how well connected your ISP is to the data center hosting that specific server.

If you are being censored I think the fact that a single server is fast and the rest is slow is that whatever is censoring you did not add this specific server's IP to its list of hosts to mess with (yet).

**Comment by logan2k** [comment]
Thank you faern.
I just tested Android on the same network, and it do not have this problem.
So it must be the Linux version issue.

**Comment by logan2k** [comment]
> Changing MTU don't help the issue.

I just saw issue #3839, maybe that's the problem?
Does #3839 also apply to 2022.2?

**Comment by faern** [comment]
Nope, that bug was only in `2022.3`. So setting MTU should work as expected on `2022.2`.
